test/e2e/certmanager.go

```diff
@@ -29,6 +29,7 @@ func runCertManagerRemoteClusterInstallSimpleFlow(test *framework.MulticlusterE2
 		packageFile := e.BuildPackageConfigFile(packageName, packagePrefix, EksaPackagesNamespace)
 		test.ManagementCluster.InstallCuratedPackageFile(packageFile, kubeconfig.FromClusterName(test.ManagementCluster.ClusterName))
 		e.VerifyCertManagerPackageInstalled(packagePrefix, EksaPackagesNamespace, cmPackageName, withMgmtClusterSetup(test.ManagementCluster))
+		e.CleanupCerts(withMgmtClusterSetup(test.ManagementCluster))
```

Comment:
Would the controller be guaranteed to have enough time to clean up the certs before the delete operation? 
Perhaps we need a wait operation before deleting?

Reply:
we're using a kubectl client ([here](https://github.com/aws/eks-anywhere/blob/35e85705aa594680bb2fde2caeb92b6f2c0b16b1/pkg/executables/kubectl.go#L2287-L2287)), so I think delete should hang until the resource is removed: https://github.com/kubernetes/kubernetes/issues/67479#issuecomment-413873372

Thoughts?

Reply:
I think the client hangs when there's a finalizer on the resource?

If the controller removes the finalizer only after cleaning up the related resources you mentioned, then yeah this looks fine to me